sdk/core/Azure.Core.Expressions.DataFactory/src/DataFactoryExpression.cs

```diff
@@ -14,21 +14,22 @@ namespace Azure.Core.Expressions.DataFactory
     /// </summary>
     /// <typeparam name="T"> Can be one of <see cref="string"/>, <see cref="bool"/>, <see cref="int"/>, <see cref="double"/>, <see cref="Array"/>. </typeparam>
 #pragma warning disable SA1649 // File name should match first type name
-    public sealed partial class DataFactoryExpression<T> : IUtf8JsonSerializable
+    [JsonConverter(typeof(DataFactoryExpressionJsonConverter))]
+    public sealed class DataFactoryExpression<T>
 #pragma warning restore SA1649 // File name should match first type name
     {
-        private string? _type;
-        private T? _literal;
-        private string? _expression;
+        internal string? Type { get; }
+        internal T? LiteralInternal { get; }
```

Comment:
I would still just call this `Literal`

Reply:
There is already a public literal property. The difference is that this property does not check `HasLiteral`